source/docs/software/telemetry/robot-telemetry-with-annotations.rst

```diff
@@ -0,0 +1,305 @@
+      - NetworkTables-based logging, which the DataLogManager can read and save to a file on disk
+    * - Root Data Entry
+      - The root entry that logged data will appear under. Used for both NetworkTables-based and DataLog-based logging. Can be set to any string.
+      - ``"Robot"``
+    * - Minimum Importance Level
+      - The minimum importance level of data for it to be logged. Any logged data marked with a lower priority will be excluded from telemetry.
+      - ``DEBUG`` (logs everything)
+    * - Error handler
+      - Catches and responds to errors encountered while logging data. Error handlers are used to prevent any errors encountered during logging from crashing the entire robot program.
+      - Errors are caught and printed to the console
+
+.. tab-set-code::
+
+    .. code-block:: java
+
+        @Logged
+        public class Robot extends TimedRobot {
+          public Robot() {
+            Epilogue.configure(config -> {
+              // Log only to disk, instead of the default NetworkTables logging
+              // Note that this means data cannot be analyzed in realtime by a dashboard
+              config.dataLogger = new FileLogger(DataLogManager.getLog());
+
+              if (isSimulation()) {
+                // If running in simulation, then we'd want to re-throw any errors that
+                // occur so we can debug and fix them!
+                config.errorHandler = ErrorHandler.crashOnError();
+              }
+
+              // Change the root data path
+              config.root = "Telemetry";
+
+              // Only log critical information instead of the default DEBUG level.
+              // This can be helpful in a pinch to reduce network bandwidth or log file size
+              // while still logging important information.
+              config.minimumImportance = Logged.Importance.CRITICAL;
+            });
+
+            Epilogue.bind(this);
+          }
+        }
+
+
+Logging Third-Party Data
+------------------------
+
+A large part of the Java ecosystem is its access to third-party libraries that provide extra functionality, such as vendor libraries for interacting with custom hardware like motor controllers and sensors. If those vendors do not use logging annotations in their libraries, then you can still log them by writing a custom logger class! These classes are actually what the WPILib annotations will generate; you just write a custom one because third-party code cannot be annotated.
+
+Custom loggers can be declared in any package, and only need to have the ``@CustomLoggerFor`` annotation present to be able to be detected and used. They must extend from ``ClassSpecificLogger`` (`Javadoc <https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/epilogue/logging/ClassSpecificLogger.html>`__) and must have a public no-argument construtor - failing to declare one will result in a compilation error - and must pass the type they log to the superclass constructor.
```

Comment:
I don't think we publish javadocs for latest/bleeding edge builds so this link will be broken until there's a release

Reply:
```suggestion
Custom loggers can be declared in any package, and only need to have the ``@CustomLoggerFor`` annotation present to be able to be detected and used. They must extend from ``ClassSpecificLogger`` (`Javadoc <https://github.wpilib.org/allwpilib/docs/development/java/edu/wpi/first/epilogue/logging/ClassSpecificLogger.html>`__) and must have a public no-argument construtor - failing to declare one will result in a compilation error - and must pass the type they log to the superclass constructor.
```